### PR TITLE
restrict program master lists when creating internal orders

### LINKED ISSUE #1697: Program orders: you can still place an "General" order for items on a "Program" master list
## What went wrong? 😲
 

once an item is on a "program" master list, you should not be able to place a general order for it

list `PLSS - ARV` is a program master list: 

Screenshot 2023-05-05 at 4 17 00 PM

but you can place a general order with it: 

Screenshot 2023-05-05 at 4 18 33 PM

## Expected behaviour 🤔

Like for CDI, for tagged stores with program master lists, you should not be able to order items through a general order 

## Your environment 🌱
 - Version 1.1.11
 - Platform macOS desktop

### PR DESCRIPTION
**Pull request by mark-prins**
Fixes #1697 

# 👩🏻‍💻 What does this PR do? 
 
Restricts the list of master lists which is shown when using a general internal order.
Adds a check to the 'addFromMasterList' mutation which also checks. This returns the 'master list not in this store' error, as I thought the event unlikely enough to warrant a new error variant. 

# 🧪 How has/should this change been tested? 

Configured multiple master lists for a store, with some program lists among them.
Check that you can create a program based internal order, and that you can add to a general order from master lists - but only lists which are not program lists.

## 📃 Documentation

- [ ] Worth mentioning that you cannot use program lists when adding to internal orders

### CODE REVIEW

File: server/repository/src/db_diesel/master_list.rs [hunk 1]
```diff
@@ -87,6 +89,19 @@ impl<'a> MasterListRepository<'a> {
 
                 query = query.filter(master_list_dsl::id.eq_any(name_join_query));
             }
+
+            if let Some(is_program) = f.is_program {
+                let program_join_query = program_dsl::program
```
Comment (andreievg) [inline server/repository/src/db_diesel/master_list.rs:94]: Nice !

**Review by andreievg — APPROVED** [review]
Looks good, but just want to clarify, we want to hide master lists that are of type program for ALL general requisition ? Even if they are not linked to the store by tag. This sounds all good, there are a couple of use case where this makes less sense (but I can't really see those use cases as very likely)

One thing that I am not a fan of is loosing type safety for graphql parameters, made an issue: https://github.com/openmsupply/open-msupply/issues/1733

**Comment by mark-prins** [comment]
> Looks good, but just want to clarify, we want to hide master lists that are of type program for ALL general requisition ?

Yeah, wasn't sure on that one. I went restrictive and just excluded all program based lists without checking to see if they're assigned. I figured that they shouldn't be available even if they aren't tagged for the current store 🤷

**Comment by mark-prins** [comment]
and thanks for review!

**Pull request merged**